backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java

```diff
@@ -283,7 +283,16 @@ private ComputingResponse getComputingMethodResponse(@NonNull final CreateInvoic
 
 		final ComputingResponse response = computingMethodHandler.compute(request);
 
-		final UomId stockUomId = productBL.getStockUOMId(moduleConfig.getProductId());
+		final UomId stockUomId;
+
+		if(moduleConfig.getModularContractType().isMatching(ComputingMethodType.INTERIM_CONTRACT))
+		{
+			stockUomId = productBL.getStockUOMId(modularContract.getM_Product_ID());
+		}
+		else
+		{
+			stockUomId = productBL.getStockUOMId(moduleConfig.getProductId());
+		}
```

Comment:
why do we need this validation?

Reply:
Added a comment in the code about validating the UOM of the contract product if interim.